pymeasure/instruments/srs/sr570.py

```diff
@@ -0,0 +1,214 @@
+        Values are truncated to the closest allowed value if not exact. """, 
+        validators=truncated_discrete_set,
+        values=OFFSET_CURRENTS,
+        map_values = True)
+    
+    offset_current_sign = Instrument.setting(
+        "IOSN %d", 
+        """ An string that sets the offset current sign. 
+        Allowed values are: 'positive' and 'negative'. """, 
+        validators=strict_discrete_set,
+        values={'positive': 1, 'negative': 0},
+        map_values=True)
+    
+    gain_mode = Instrument.setting(
+        "GNMD %d", 
+        """ A string that sets the gain mode.
+        Allowed values are: {}""".format(GAIN_MODES), 
+        validators=truncated_discrete_set,
+        values = GAIN_MODES,
+        map_values=True)
+    
+    invert_signal_sign = Instrument.setting(
+        "INVT %d", 
+        """ An boolean sets the signal invert sense.
+        Allowed values are: True (inverted) and False (not inverted). """, 
+        validators=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True)
+    
+    bias_enabled = Instrument.setting(
+        "BSON %d", 
+        """ Boolean that turns the bias on or off.
+        Allowed values are: True (bias on) and False (bias off)""", 
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True)
+    
+    offset_current_enabled = Instrument.setting(
+        "IOON %d", 
+        """ Boolean that turns the offset current on or off.
+        Allowed values are: True (current on) and False (current off).""", 
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True)
+    
+    front_blanked = Instrument.setting(
+        "BLNK %d", 
+        """ Boolean that blanks(True) or un-blanks (False) the front panel""", 
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True)
+    
+    signal_inverted = Instrument.setting(
+        "INVT %d", 
+        """ Boolean that inverts the signal if True""", 
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True)
+    
+    ####################
+    # Methods        #
+    ####################
+    
+    def enable_bias(self):
+        """Turns the bias voltage on"""
+        self.bias_enabled = True
+    
+    def disable_bias(self):
+        """Turns the bias voltage off"""
+        self.bias_enabled = False
+    
+    def enable_offset_current(self):
+        """"Enables the offset current """
+        self.offset_current_enabled = True
+        
+    def disable_offset_current(self):
+        """"Disables the offset current """
+        self.offset_current_enabled = False
+    
+    def clear_overload(self):
+        """"Reset the filter capacitors to clear an overload condition"""
+        self.write("ROLD")
+    
+    def reset_instrument(self):
+        """Resets the amplifier to default settings"""
+        self.write('*RST')
```

Comment:
A method for sending `*RST` is standard implemented (named `reset`), so you can remove this to prevent code duplication.

Reply:
Ah yes I see, I did not check this. This is fixed now, thanks for your thorough reviewing work :)